### PR TITLE
Merge feature/37-name

### LINKED ISSUE #37: Run two vagrant boxes simultaniously
At the moment it is not possible to run two vagrant boxes at the same time.

That’s because when running `vagrant up` (second step) vagrant tries to rename a temp folder located in user/VirtualBox VMs/ to joomlatools-box which already exists.

Note: you can solve this by just allowing the box to have a different name. We solved this for nooku-server box too. See: https://github.com/nooku/nooku-server/blob/develop/Vagrantfile.pkg#L30-L39 (see all commits that mention ticket #281)

### PR DESCRIPTION
**Pull request by stevenrombauts**
Closes #37

### CODE REVIEW

**Comment by stevenrombauts** [comment]
@yiendos Could you give this a go please on your machine? Testing steps are as follows:

1. Clone the repository in a separate location:

    ```
    cd /tmp/
    git clone --depth 1 -b feature/37-name git@github.com:joomlatools/joomlatools-vagrant.git
    ```

1. Copy the custom config file:

    ```
    cp config.custom.yaml-dist config.custom.yaml
    ```

1. Edit the `config.custom.yaml` file and remove the `synced_folders` entry (if the path is nonexistent, you'll get errors)
1. Change the `name`, `ip` and/or any of the other `virtualbox` settings in `config.custom.yaml`
1. Run `vagrant up`
1. Open the Virtualbox application and verify that the new box is running alongside your existing box.

**Comment by yiendos** [comment]
@stevenrombauts apologises for the delay, I can confirm I can carry out all the steps listed above, and that the new box can quite happily sit alongside my existing installation. 

I am getting quite a few build errors however: 
```
Profiles::Box::Cli]
==> default: Error: /Stage[main]/Profiles::Cloudcommander/Package[cloudcmd]: Provider npm is not functional on this host
==> default: Error: /Stage[main]/Profiles::Box::Tools/Package[yarn]: Provider npm is not functional on this host
==> default: Error: /Stage[main]/Profiles::Box::Tools/Package[less]: Provider npm is not functional on this host
==> default: Error: /Stage[main]/Profiles::Box::Tools/Package[autoless]: Provider npm is not functional on this host
==> default: Error: /Stage[main]/Profiles::Box::Tools/Package[uglify-js]: Provider npm is not functional on this host
==> default: Notice: /Stage[main]/Mysql::Server::Installdb/File[/var/log/mysql/error.log]: Dependency Mysql_datadir[/var/lib/mysql] has failures: true
==> default: Warning: /Stage[main]/Mysql::Server::Installdb/File[/var/log/mysql/error.log]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Mysql::Server::Service/Service[mysqld]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Mysql::Server::Service/Exec[wait_for_mysql_socket_to_open]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Mysql::Server::Root_password/Exec[remove install pass]: Skipping because of failed dependencies
==> default: Notice: /Stage[main]/Profiles::Wetty/Exec[yarn-install-wetty]: Dependency Package[yarn] has failures: true
==> default: Warning: /Stage[main]/Profiles::Wetty/Exec[yarn-install-wetty]: Skipping because of failed dependencies
==> default: Notice: /Stage[main]/Profiles::Cloudcommander/File[/lib/systemd/system/cloudcommander.service]: Dependency Package[cloudcmd] has failures: true
==> default: Warning: /Stage[main]/Profiles::Cloudcommander/File[/lib/systemd/system/cloudcommander.service]: Skipping because of failed dependencies
==> default: Info: Class[Profiles::Systemd::Reload]: Unscheduling all events on Class[Profiles::Systemd::Reload]
==> default: Warning: /Stage[main]/Profiles::Systemd::Reload/Exec[/bin/systemctl daemon-reload]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Php/Service[php-fpm]: Skipping because of failed dependencies
==> default: Info: /Stage[main]/Profiles::Php/Service[php-fpm]: Unscheduling all events on Service[php-fpm]
==> default: Info: Class[Profiles::Php]: Unscheduling all events on Class[Profiles::Php]
==> default: Warning: /Stage[main]/Profiles::Phpmyadmin/File[/usr/share/phpmyadmin]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Phpmyadmin/Exec[install-phpmyadmin]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Phpmyadmin/File[/usr/share/phpmyadmin/config.inc.php]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Phpmyadmin/Exec[creating-phpmyadmin-controluser]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Mailhog/Service[mailhog]: Skipping because of failed dependencies
==> default: Info: /Stage[main]/Profiles::Mailhog/Service[mailhog]: Unscheduling all events on Service[mailhog]
==> default: Info: Class[Profiles::Mailhog]: Unscheduling all events on Class[Profiles::Mailhog]
==> default: Warning: /Stage[main]/Profiles::Wetty/Service[wetty]: Skipping because of failed dependencies
==> default: Info: /Stage[main]/Profiles::Wetty/Service[wetty]: Unscheduling all events on Service[wetty]
==> default: Info: Class[Profiles::Wetty]: Unscheduling all events on Class[Profiles::Wetty]
==> default: Warning: /Stage[main]/Profiles::Cloudcommander/Service[cloudcommander]: Skipping because of failed dependencies
==> default: Info: /Stage[main]/Profiles::Cloudcommander/Service[cloudcommander]: Unscheduling all events on Service[cloudcommander]
==> default: Info: Class[Profiles::Cloudcommander]: Unscheduling all events on Class[Profiles::Cloudcommander]
==> default: Info: Class[Profiles::Box::Tools]: Unscheduling all events on Class[Profiles::Box::Tools]
==> default: Warning: /Stage[main]/Profiles::Box/File[/etc/apache2/joomla.box-include.conf]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Box/Apache::Vhost[joomla.box-http]/File[/var/www]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Phpmyadmin/Apache::Vhost[phpmyadmin.joomla.box]/Concat[10-phpmyadmin.joomla.box.conf]/Concat_file[10-phpmyadmin.joomla.box.conf]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Phpmyadmin/Apache::Vhost[phpmyadmin.joomla.box]/Concat[10-phpmyadmin.joomla.box.conf]/File[/etc/apache2/sites-available/10-phpmyadmin.joomla.box.conf]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Phpmyadmin/Apache::Vhost[phpmyadmin.joomla.box]/File[10-phpmyadmin.joomla.box.conf symlink]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Phpmyadmin/Apache::Vhost[phpmyadmin.joomla.box]/Concat::Fragment[phpmyadmin.joomla.box-apache-header]/Concat_fragment[phpmyadmin.joomla.box-apache-header]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Phpmyadmin/Apache::Vhost[phpmyadmin.joomla.box]/Concat::Fragment[phpmyadmin.joomla.box-docroot]/Concat_fragment[phpmyadmin.joomla.box-docroot]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Phpmyadmin/Apache::Vhost[phpmyadmin.joomla.box]/Concat::Fragment[phpmyadmin.joomla.box-directories]/Concat_fragment[phpmyadmin.joomla.box-directories]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Phpmyadmin/Apache::Vhost[phpmyadmin.joomla.box]/Concat::Fragment[phpmyadmin.joomla.box-logging]/Concat_fragment[phpmyadmin.joomla.box-logging]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Phpmyadmin/Apache::Vhost[phpmyadmin.joomla.box]/Concat::Fragment[phpmyadmin.joomla.box-serversignature]/Concat_fragment[phpmyadmin.joomla.box-serversignature]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Phpmyadmin/Apache::Vhost[phpmyadmin.joomla.box]/Concat::Fragment[phpmyadmin.joomla.box-access_log]/Concat_fragment[phpmyadmin.joomla.box-access_log]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Phpmyadmin/Apache::Vhost[phpmyadmin.joomla.box]/Concat::Fragment[phpmyadmin.joomla.box-file_footer]/Concat_fragment[phpmyadmin.joomla.box-file_footer]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Box/Apache::Vhost[joomla.box-http]/Concat[00-joomla.box-http.conf]/Concat_file[00-joomla.box-http.conf]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Box/Apache::Vhost[joomla.box-http]/Concat[00-joomla.box-http.conf]/File[/etc/apache2/sites-available/00-joomla.box-http.conf]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Box/Apache::Vhost[joomla.box-http]/File[00-joomla.box-http.conf symlink]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Box/Apache::Vhost[joomla.box-http]/Concat::Fragment[joomla.box-http-apache-header]/Concat_fragment[joomla.box-http-apache-header]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Box/Apache::Vhost[joomla.box-http]/Concat::Fragment[joomla.box-http-docroot]/Concat_fragment[joomla.box-http-docroot]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Box/Apache::Vhost[joomla.box-http]/Concat::Fragment[joomla.box-http-directories]/Concat_fragment[joomla.box-http-directories]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Box/Apache::Vhost[joomla.box-http]/Concat::Fragment[joomla.box-http-additional_includes]/Concat_fragment[joomla.box-http-additional_includes]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Box/Apache::Vhost[joomla.box-http]/Concat::Fragment[joomla.box-http-logging]/Concat_fragment[joomla.box-http-logging]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Box/Apache::Vhost[joomla.box-http]/Concat::Fragment[joomla.box-http-serversignature]/Concat_fragment[joomla.box-http-serversignature]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Box/Apache::Vhost[joomla.box-http]/Concat::Fragment[joomla.box-http-access_log]/Concat_fragment[joomla.box-http-access_log]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Box/Apache::Vhost[joomla.box-http]/Concat::Fragment[joomla.box-http-serveralias]/Concat_fragment[joomla.box-http-serveralias]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Box/Apache::Vhost[joomla.box-http]/Concat::Fragment[joomla.box-http-file_footer]/Concat_fragment[joomla.box-http-file_footer]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Box/Apache::Vhost[joomla.box-ssl]/Concat[00-joomla.box-ssl.conf]/Concat_file[00-joomla.box-ssl.conf]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Box/Apache::Vhost[joomla.box-ssl]/Concat[00-joomla.box-ssl.conf]/File[/etc/apache2/sites-available/00-joomla.box-ssl.conf]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Box/Apache::Vhost[joomla.box-ssl]/File[00-joomla.box-ssl.conf symlink]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Apache::Service/Service[httpd]: Skipping because of failed dependencies
==> default: Info: /Stage[main]/Apache::Service/Service[httpd]: Unscheduling all events on Service[httpd]
==> default: Warning: /Stage[main]/Profiles::Varnish/Service[varnish]: Skipping because of failed dependencies
==> default: Info: /Stage[main]/Profiles::Varnish/Service[varnish]: Unscheduling all events on Service[varnish]
==> default: Info: Class[Profiles::Varnish]: Unscheduling all events on Class[Profiles::Varnish]
==> default: Warning: /Stage[main]/Profiles::Box/Apache::Vhost[joomla.box-ssl]/Concat::Fragment[joomla.box-ssl-apache-header]/Concat_fragment[joomla.box-ssl-apache-header]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Box/Apache::Vhost[joomla.box-ssl]/Concat::Fragment[joomla.box-ssl-docroot]/Concat_fragment[joomla.box-ssl-docroot]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Box/Apache::Vhost[joomla.box-ssl]/Concat::Fragment[joomla.box-ssl-directories]/Concat_fragment[joomla.box-ssl-directories]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Box/Apache::Vhost[joomla.box-ssl]/Concat::Fragment[joomla.box-ssl-additional_includes]/Concat_fragment[joomla.box-ssl-additional_includes]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Box/Apache::Vhost[joomla.box-ssl]/Concat::Fragment[joomla.box-ssl-logging]/Concat_fragment[joomla.box-ssl-logging]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Box/Apache::Vhost[joomla.box-ssl]/Concat::Fragment[joomla.box-ssl-serversignature]/Concat_fragment[joomla.box-ssl-serversignature]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Box/Apache::Vhost[joomla.box-ssl]/Concat::Fragment[joomla.box-ssl-access_log]/Concat_fragment[joomla.box-ssl-access_log]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Box/Apache::Vhost[joomla.box-ssl]/Concat::Fragment[joomla.box-ssl-serveralias]/Concat_fragment[joomla.box-ssl-serveralias]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Box/Apache::Vhost[joomla.box-ssl]/Concat::Fragment[joomla.box-ssl-ssl]/Concat_fragment[joomla.box-ssl-ssl]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Box/Apache::Vhost[joomla.box-ssl]/Concat::Fragment[joomla.box-ssl-file_footer]/Concat_fragment[joomla.box-ssl-file_footer]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Phpmyadmin/Apache::Vhost[phpmyadmin.joomla.box]/Apache::Listen[80]/Concat::Fragment[Listen 80]/Concat_fragment[Listen 80]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Profiles::Box/Apache::Vhost[joomla.box-ssl]/Apache::Listen[443]/Concat::Fragment[Listen 443]/Concat_fragment[Listen 443]: Skipping because of failed dependencies
==> default: Notice: /Stage[main]/Mysql::Server::Root_password/File[/root/.my.cnf]: Dependency Mysql_user[root@localhost] has failures: true
==> default: Warning: /Stage[main]/Mysql::Server::Root_password/File[/root/.my.cnf]: Skipping because of failed dependencies
==> default: Notice: /Stage[main]/Mysql::Server/Anchor[mysql::server::end]: Dependency Mysql_user[root@%] has failures: true
==> default: Notice: /Stage[main]/Mysql::Server/Anchor[mysql::server::end]: Dependency Mysql_grant[root@%/*.*] has failures: true
==> default: Warning: /Stage[main]/Mysql::Server/Anchor[mysql::server::end]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Mysql::Client/Anchor[mysql::client::start]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Mysql::Client::Install/Package[mysql_client]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Mysql::Client/Anchor[mysql::client::end]: Skipping because of failed dependencies
==> default: Info: Stage[main]: Unscheduling all events on Stage[main]
==> default: Error: Could not find a suitable provider for mysql_datadir
==> default: Error: Could not find a suitable provider for mysql_user
==> default: Error: Could not find a suitable provider for mysql_grant
==> default: Info: Creating state file /opt/puppetlabs/puppet/cache/state/state.yaml
==> default: Notice: Applied catalog in 29.04 seconds
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```

File: Vagrantfile.pkg [hunk 1]
```diff
@@ -13,7 +13,15 @@ _config = {
         "/var/www" => File.join(Dir.pwd, "www"),
```
Comment (yiendos) [inline Vagrantfile.pkg]: @stevenrombauts I know about the vagrantfile, but what does the vagrantfile.pkg do?

**Comment by yiendos** [comment]
@stevenrombauts ignore my comments about the box not installing, I've had these problems before and require me to:
```

`touch /tmp/mysql.key`

`nano /tmp/mysql.key` 

copy paste of mysql.key 

`sudo apt-key add /tmp/mysql.key`
```

File: Vagrantfile.pkg (hunk 1 above)
Comment (stevenrombauts) [reply Vagrantfile.pkg]: @yiendos The `.pkg` file is the Vagrantfile that we package with the published box. So the actual `Vagrantfile` is only used when you checkout the repository and run it yourself. The `Vagrantfile.pkg` is what Vagrant will use if you download the box (using `vagrant init joomlatools/box`

Comment (stevenrombauts) [reply Vagrantfile.pkg]: It's used in the Packer build instructions here: https://github.com/joomlatools/joomlatools-vagrant/blob/master/packer.json#L102-L107

**Comment by stevenrombauts** [comment]
@yiendos Those errors do sound alarming. This box should build itself in a single go. These warnings are very strange: 

> Provider npm is not functional on this host

I'll build again from scratch when I do the final preparations in another ticket. Can you confirm the change from this PR works or are you still having problems?

**Comment by yiendos** [comment]
@stevenrombauts happy to confirm that we could have two boxes running side by side thanks to the configuration work that you have done.

**Comment by stevenrombauts** [comment]
Thanks @yiendos

**Pull request merged**